doc/java.md

```diff
@@ -0,0 +1,298 @@
+BR_LOAD_ACC_CONV_5:
+
+    aload       ACC_CONV_5_LOAD
+    invokestatic    #Integer.value
+    istore      ACC_CONV_5              // acc_conv_5 = acc_conv_5_load.value()
+
+BR_SKIP_ACC_CONV_5:
+
+    aload       ICICLE
+    invokevirtual   #IcicleMagic.StartNew
+
+BR_WHILE_START:
+
+    aload       ICICLE
+    invokevirtual   #IcicleMagic.HasField
+    ifeq        BR_WHILE_END        // if (icicle.HasField() == 0) goto end
+
+
+    aload       ICICLE
+    invokevirtual   #IcicleMagic.CurrentRow
+    checkcast       #Integer
+    invokevirtual   #Integer.value
+    istore      ANF_3               // anf_3 = icicle.CurrentRow<Integer>().value;
+
+    iload       ANF_3
+    iconst      10
+    if_icmpgt   BR_ANF_3_THEN       // if (anf_3 > 10)
+    goto        BR_ANF_3_ELSE
+
+BR_ANF_3_THEN:
+    iload       ACC_CONV_5
+    istore      READ_ACC_CONV_5     // read_acc_conv_5 = acc_conv_5;
+
+    iload       READ_ACC_CONV_5
+    iconst      1
+    iadd
+    istore      ACC_CONV_5          // acc_conv_5 = read_acc_conv_5 + 1
+
+BR_ANF_3_ELSE:
+    goto BR_WHILE_START
+
+
+BR_WHILE_END:
+
+    aload       ICICLE
+    ldc         "CountAboveTen"
+    ldc         "acc$conv$5"
+
+    new         #Integer
+    dup
+    iload       ACC_CONV_5
+    invokespecial #Integer.<init>
+
+    invokestatic #IcicleMagic.SaveResumable // icicle.SaveResumable<Integer>(...);
+
+    iload       ACC_CONV_5
+    istore      CONV_5
+
+    aload       ICICLE
+    ldc         "CountAboveTen"
+
+    new         #Integer
+    dup
+    iload       ACC_CONV_5
+    invokespecial #Integer.<init>
+
+    invokestatic #IcicleMagic.Output    // icicle.Output<Integer>("CountAboveTen", new Integer(conv_5));
+}
+```
+
+
+CountLastWeek
+-------------
+
+How many things happened last week?
+Input query:
+```
+feature salary ~> windowed between 7 days and 14 days ~> count
+```
+
+Flattened Avalanche.
+```
+gen$date = DATE
+{
+  init acc$conv$4 = 0 : Int (Windowed);
+  for_facts (gen$fact : Int) in history {
+    let anf$4 = snd# [Int] [DateTime] gen$fact;
+    let anf$5 = DateTime_daysDifference# anf$4 gen$date;
+    let anf$6 = le# [Int] anf$5 (14 : Int);
+    let anf$8 = ge# [Int] anf$5 (7 : Int);
+    if (and# anf$6 anf$8) {
+      read acc$conv$4 = acc$conv$4;
+      write acc$conv$4 = add# acc$conv$4 (1 : Int);
+    } else {
+      if (lt# [Int] anf$5 (7 : Int)) {
+        keep_fact_in_history;
+      }
+
+    }
+
+  }
+  for_facts (gen$fact : Int) in new {
+    let anf$16 = snd# [Int] [DateTime] gen$fact;
+    let anf$17 = DateTime_daysDifference# anf$16 gen$date;
+    let anf$18 = le# [Int] anf$17 (14 : Int);
+    let anf$20 = ge# [Int] anf$17 (7 : Int);
+    if (and# anf$18 anf$20) {
+      read acc$conv$4 = acc$conv$4;
+      write acc$conv$4 = add# acc$conv$4 (1 : Int);
+    } else {
+      if (lt# [Int] anf$17 (7 : Int)) {
+        keep_fact_in_history;
+      }
+
+    }
+
+  }
+  read conv$4 = acc$conv$4;
+  return conv$4;
+}
+```
+
+Java:
+```
+void Compute(IcicleMagic icicle)
```

Comment:
Looks good to me. Charles pointed out the convention of lower camel case for method names which is different to what you have. (Curiously C# uses upper camel case for method names, mostly out of spite)

(and looks like you have switched $ for _ which seems more idiomatic)